src/Framework/FileAccess/ReportedFileOperation.cs

```diff
@@ -0,0 +1,257 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Build.Framework.FileAccess
+{
+    /// <summary>
+    /// Which operation resulted in a reported file access.
+    /// </summary>
+    public enum ReportedFileOperation : byte
+    {
+        /// <summary>
+        /// Unknown operation.
+        /// </summary>
+        Unknown = 0,
+
+        /// <summary>
+        /// CreateFile.
+        /// </summary>
+        CreateFile,
+
+        /// <summary>
+        /// CreateProcess.
+        /// </summary>
+        CreateProcess,
+
+        /// <summary>
+        /// GetFileAttributes.
+        /// </summary>
+        GetFileAttributes,
```

Comment:
It's up to the plugin to determine what it cares about. For example, for QuckBuild (and my prototype plugin), I currently don't care about probes. But BXL does, and that's the more correct thing to do.

Also, if you do care about probes, then the file existance is what matters there, while for reads the file content matters.

It can also be helpful to know whether the read/write is a copy or not since some optimizations can (in theory) be made if we know information about the source file and can infer that same information about the destination (assuming no other writes to the destination).